script/Deployer.s.sol

```diff
@@ -0,0 +1,275 @@
+
+        (address compVotingStrategy, ) = noRedeploy(deployer, type(CompVotingStrategy).creationCode, saltNonce);
+        deployments.serialize("CompVotingStrategy", compVotingStrategy);
+
+        (address ozVotesVotingStrategy, ) = noRedeploy(deployer, type(OZVotesVotingStrategy).creationCode, saltNonce);
+        deployments.serialize("OZVotesVotingStrategy", ozVotesVotingStrategy);
+
+        (address merkleWhitelistVotingStrategy, ) = noRedeploy(
+            deployer,
+            type(MerkleWhitelistVotingStrategy).creationCode,
+            saltNonce
+        );
+        deployments.serialize("MerkleWhitelistVotingStrategy", merkleWhitelistVotingStrategy);
+
+        // ------- PROPOSAL VALIDATION STRATEGIES -------
+
+        (address vanillaProposalValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(VanillaProposalValidationStrategy).creationCode,
+            saltNonce
+        );
+        deployments.serialize("VanillaProposalValidationStrategy", vanillaProposalValidationStrategy);
+
+        (address propositionPowerProposalValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(PropositionPowerProposalValidationStrategy).creationCode,
+            saltNonce
+        );
+        deployments.serialize("PropositionPowerProposalValidationStrategy", propositionPowerProposalValidationStrategy);
+
+        (address activeProposalsLimiterProposalValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(ActiveProposalsLimiterProposalValidationStrategy).creationCode,
+            saltNonce
+        );
+
+        deployments.serialize(
+            "ActiveProposalsLimiterProposalValidationStrategy",
+            activeProposalsLimiterProposalValidationStrategy
+        );
+
+        (address propositionPowerAndActiveProposalsLimiterValidationStrategy, ) = noRedeploy(
+            deployer,
+            type(PropositionPowerAndActiveProposalsLimiterValidationStrategy).creationCode,
+            saltNonce
+        );
+
+        deployments.serialize(
+            "PropositionPowerAndActiveProposalsLimiterValidationStrategy",
+            propositionPowerAndActiveProposalsLimiterValidationStrategy
+        );
+
+        // ------- PROXY FACTORY -------
+
+        (address proxyFactory, ) = noRedeploy(deployer, type(ProxyFactory).creationCode, saltNonce);
+        deployments.serialize("ProxyFactory", proxyFactory);
+
+        // ------- SPACE -------
+
+        (address space, ) = noRedeploy(deployer, type(Space).creationCode, saltNonce);
+
+        // If the master space is not initialized, initialize it
+        if (Space(space).owner() == address(0x0)) {
+            // Initializer for the master space, to render it unusable
+            Strategy[] memory emptyStrategyArray = new Strategy[](1);
+            emptyStrategyArray[0] = Strategy(address(0x1), new bytes(0));
+            string[] memory emptyStringArray = new string[](1);
```

Comment:
Might be worth renaming `empty` to maybe `default` or `unusable` or `test` or `Dummy` ? (I got confused why [this code](https://github.com/snapshot-labs/sx-evm/blob/4a693e4adee064c2dea42b44719bb4a860fe3fe6/src/Space.sol#L77-L79) would not trigger if we were passing empty arrays :p)

Reply:
updated